In [1]:
%matplotlib inline

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [24]:
activities = pd.read_csv('data/atusact_2014.dat')
activities = activities[['TUCASEID', 'TUACTIVITY_N', 'TUTIER1CODE', 'TUTIER2CODE', 'TUTIER3CODE', 'TRCODE', 'TRTIER2', 'TUACTDUR']]
activities.head()

TUCASEID  TUACTIVITY_N  TUTIER1CODE  TUTIER2CODE  TUTIER3CODE  \
0  20140101140007             1            1            1            1   
1  20140101140007             2            3            1            1   
2  20140101140007             3            1            2            1   
3  20140101140007             4           18           11            1   
4  20140101140007             5           11            1            1   

   TRCODE  TRTIER2  TUACTDUR  
0   10101      101       540  
1   30101      301        30  
2   10201      102        60  
3  181101     1811        10  
4  110101     1101        30

In [45]:
sleepers = activities[activities.TRCODE == 10101]
sleepers.head()

TUCASEID  TUACTIVITY_N  TUTIER1CODE  TUTIER2CODE  TUTIER3CODE  \
0   20140101140007             1            1            1            1   
9   20140101140007            10            1            1            1   
10  20140101140011             1            1            1            1   
18  20140101140011             9            1            1            1   
27  20140101140011            18            1            1            1   

    TRCODE  TRTIER2  TUACTDUR  
0    10101      101       540  
9    10101      101       725  
10   10101      101       150  
18   10101      101       120  
27   10101      101       420

In [47]:
act_sum = pd.read_csv('data/atussum_2014.dat')
act_sum.head()

tucaseid     TUFINLWGT  TRYHHCHILD  TEAGE  TESEX  PEEDUCA  PTDTRACE  \
0  20140101140007  4.891316e+06           0     20      2       39         1   
1  20140101140011  2.314737e+07          16     16      2       37         2   
2  20140101140028  2.094182e+07           2     65      2       38         1   
3  20140101140063  2.505899e+06          -1     50      2       39         2   
4  20140101140168  1.592823e+07           8     23      2       39         2   

   PEHSPNON  GTMETSTA  TELFS   ...     t181601  t181699  t181801  t181899  \
0         1         1      4   ...           0        0        0        0   
1         2         1      1   ...           0        0        0        0   
2         1         1      5   ...           0        0        0        0   
3         2         2      5   ...           0        0        0        0   
4         2         2      4   ...           0        0        0        0   

   t189999  t500101  t500103  t500105  t500106  t500107  
0        0        0        0        0        0        0  
1        0        0        0        0        0        0  
2        0        0        0        0        0        0  
3        0        0        0        0        0        0  
4        0        0        0        0        0        0  

[5 rows x 409 columns]

In [51]:
F_data = act_sum[act_sum.TESEX == 2]
M_data = act_sum[act_sum.TESEX == 1]
F_data.head()

tucaseid     TUFINLWGT  TRYHHCHILD  TEAGE  TESEX  PEEDUCA  PTDTRACE  \
0  20140101140007  4.891316e+06           0     20      2       39         1   
1  20140101140011  2.314737e+07          16     16      2       37         2   
2  20140101140028  2.094182e+07           2     65      2       38         1   
3  20140101140063  2.505899e+06          -1     50      2       39         2   
4  20140101140168  1.592823e+07           8     23      2       39         2   

   PEHSPNON  GTMETSTA  TELFS   ...     t181601  t181699  t181801  t181899  \
0         1         1      4   ...           0        0        0        0   
1         2         1      1   ...           0        0        0        0   
2         1         1      5   ...           0        0        0        0   
3         2         2      5   ...           0        0        0        0   
4         2         2      4   ...           0        0        0        0   

   t189999  t500101  t500103  t500105  t500106  t500107  
0        0        0        0        0        0        0  
1        0        0        0        0        0        0  
2        0        0        0        0        0        0  
3        0        0        0        0        0        0  
4        0        0        0        0        0        0  

[5 rows x 409 columns]

In [124]:
eating = activities[activities.TUTIER1CODE == 11]
eat_len = eating.groupby('TUCASEID').sum()
# eat_len = eat_len['TUACTDUR']
eat_len.head()

TUACTIVITY_N  TUTIER1CODE  TUTIER2CODE  TUTIER3CODE  TRCODE  \
TUCASEID                                                                      
20140101140007             5           11            1            1  110101   
20140101140011            19           22            2            2  220202   
20140101140028             3           11            1            1  110101   
20140101140168            29           33            3            3  330303   
20140101140537             3           11            1            1  110101   

                TRTIER2  TUACTDUR  
TUCASEID                           
20140101140007     1101        30  
20140101140011     2202        45  
20140101140028     1101        60  
20140101140168     3303        65  
20140101140537     1101        30

In [73]:
len(eat_len)

11062

In [74]:
len(act_sum)

11592

In [76]:
respondent = pd.read_csv('data/atusresp_2014.dat')
respondent.head()

TUCASEID  TULINENO  TUYEAR  TUMONTH  TEABSRSN  TEERN  TEERNH1O  \
0  20140101140007         1    2014        1        -1     -1        -1   
1  20140101140011         1    2014        1        -1     -1        -1   
2  20140101140028         1    2014        1        -1     -1        -1   
3  20140101140063         1    2014        1        -1     -1        -1   
4  20140101140168         1    2014        1        -1     -1        -1   

   TEERNH2  TEERNHRO  TEERNHRY   ...     TXSPEMPNOT  TXSPUHRS  TXTCC  \
0       -1        -1        -1   ...              0         0      0   
1      500        -1         1   ...             -1        -1     -1   
2       -1        -1        -1   ...             -1        -1      0   
3       -1        -1        -1   ...             -1        -1     -1   
4       -1        -1        -1   ...             -1        -1      0   

   TXTCCTOT  TXTCOC  TXTHH  TXTNOHH  TXTO  TXTOHH  TXTONHH  
0         0       0      0       -1     0       0       -1  
1         0       0     -1       -1    -1      -1       -1  
2         0       0      0        0    -1      -1       -1  
3         0       0     -1       -1    -1      -1       -1  
4         0       0      0       -1     0       0       -1  

[5 rows x 175 columns]

In [77]:
len(respondent)

11592

In [79]:
roster = pd.read_csv('data/atusrost_2014.dat')
roster.head()

TUCASEID  TULINENO  TEAGE  TERRP  TESEX  TXAGE  TXRRP  TXSEX
0  20140101140007         1     20     19      2      0      0      0
1  20140101140007         2     19     21      1      0      0      0
2  20140101140007         3      0     22      1      0      0      0
3  20140101140011         1     16     18      2      0      0      0
4  20140101140011         2     33     24      2      0      0      0

In [104]:
roster = roster[roster['TULINENO'] == 1]
len(roster)

11592

In [122]:
people_eating = roster.join(eat_len, on='TUCASEID')
# people_eating['min_eating'] = eat_len['TUACTDUR']
people_eating.head()

TUCASEID  TULINENO  TEAGE  TERRP  TESEX  TXAGE  TXRRP  TXSEX  \
0   20140101140007         1     20     19      2      0      0      0   
3   20140101140011         1     16     18      2      0      0      0   
5   20140101140028         1     65     18      2      0      0      0   
11  20140101140063         1     50     19      2      0      0      0   
12  20140101140168         1     23     18      2      0      0      0   

    TUACTDUR  
0       30.0  
3       45.0  
5       60.0  
11       NaN  
12      65.0

In [131]:
w_eat = people_eating[people_eating.TESEX == 2].mean().TUACTDUR
m_eat = people_eating[people_eating.TESEX == 1].mean().TUACTDUR
print(w_eat, m_eat)

67.1008253763 71.5846815482


Women spent an average of 67.1 minutes a day eating. Men spent an average of 71.58 minutes a day eating.